File: objectiveai-py/objectiveai/swarm/http.py
"""HTTP functions for swarm endpoints."""

from __future__ import annotations

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from objectiveai.client import ObjectiveAI
    from objectiveai.swarm import GetSwarm, ListSwarm, UsageSwarm


async def list_swarms(client: ObjectiveAI) -> ListSwarm:
    """List all swarms that have been used."""
    return await client.get_unary("swarms")


async def get_swarm(client: ObjectiveAI, swarm_id: str) -> GetSwarm:
    """Retrieve a specific swarm by its content-addressed ID."""
    return await client.get_unary(f"swarms/{swarm_id}")


async def get_swarm_usage(
    client: ObjectiveAI, swarm_id: str,
) -> UsageSwarm:
    """Retrieve usage statistics for a specific swarm."""
    return await client.get_unary(f"swarms/{swarm_id}/usage")

File: objectiveai-rs/src/swarm/http.rs
//! HTTP client functions for Swarm endpoints.

use crate::{HttpClient, HttpError};

/// Lists all Swarms that have been used.
pub async fn list_swarms(
    client: &HttpClient,
) -> Result<super::response::ListSwarm, HttpError> {
    client
        .send_unary(reqwest::Method::GET, "swarms", None::<String>)
        .await
}

/// Retrieves a specific Swarm by its content-addressed ID.
pub async fn get_swarm(
    client: &HttpClient,
    swarm_id: &str,
) -> Result<super::response::GetSwarm, HttpError> {
    client
        .send_unary(
            reqwest::Method::GET,
            &format!("swarms/{}", swarm_id),
            None::<String>,
        )
        .await
}

/// Retrieves usage statistics for a specific Swarm.
pub async fn get_swarm_usage(
    client: &HttpClient,
    swarm_id: &str,
) -> Result<super::response::UsageSwarm, HttpError> {
    client
        .send_unary(
            reqwest::Method::GET,
            &format!("swarms/{}/usage", swarm_id),
            None::<String>,
        )
        .await
}
